TestBenches/InputRouter_test.cpp

```diff
@@ -230,82 +393,123 @@ void prepareInputStreams( ifstream * pInputStreams
+    std::cout << "Layer# " << hLyrId 
+      << "[Index=" << hLyrId << "]"
       << " " << cNPhiBns 
       << " phi bins"
+      << " barrel bit is set to "
+      << +hIsBrl
       << "\n";
-    
+    int cLyrCorr = (hIsBrl==1) ? 0 : 10; 
     for( int cPhiBn=0; cPhiBn<cNPhiBns; cPhiBn++)
     {
-      std::string cMemPrint = getMemPrintName(cDTCName ,cLyrIndx, cPhiBn, pNonant, hLinkWord);
-      if( !cMemPrint.empty() )
-      {
-        std::cout << "Loading stubs from .. " << cMemPrint << "\n";
-        openDataFile(pInputStreams[cMemIndx],cMemPrint); 
+      std::string cFileName = "emData/MemPrints/InputStubs/"; 
+      std::string cMemPrint = getMemPrintName( cDtcName, pDTCsplit, pNonant, hLyrId + cLyrCorr, cPhiBn ); 
+      cFileName += cMemPrint;
+
+      std::ifstream fin_mem_prints;
+      bool cPrintError=false;
+      bool cFileFnd = openDataFile(fin_mem_prints,cFileName,cPrintError);
+      if( cFileFnd ){ 
+        fin_mem_prints.close();
+        std::cout << "\t..Loading stubs from .. " << cFileName 
+          << " [Mem# " << +cMemIndx << "]\n";
+        openDataFile(pInputStreams[cMemIndx],cFileName); 
+        cMemIndx++;
       }
-      cMemIndx++;
     }
   }
 }
 
+
 // test bench starts here 
-int main()
+int main(int argc, char * argv[])
 {
-  
-  // allow for truncation memory check [i.e. missing entries can pass check]
-  bool cTruncation=false;
+  // default values for test bench 
+  int cLinkId = 6; 
   int cDTCsplit=0;
   int cNonant=4;
-  std::string cInputFile_LinkMap = "emData/dtclinklayerdisk.dat";
-  std::string cBaseName  = "emData/MemPrints/InputStubs/";
-  
+  // if cmd line args are passed 
+  // parse them and change defaults 
+  if( argc > 1 )
+  {
+    std::cout << "Have passed " << +argc << " arguments to the IR test bench.\n";
+    for( size_t cArgIndx=1; cArgIndx < (size_t)argc ; cArgIndx++)
+    {
+      std::string cArg = argv[cArgIndx];
+      if( cArg.find("link") != std::string::npos ) 
+      {
+        cLinkId  = std::stol( cArg.substr( cArg.find(",")+1 , cArg.length()-1) , nullptr, 10 );
+        std::cout << "Link id passed from cmd line is " << cLinkId << "\n";
+      }
+      if( cArg.find("dtcSplit") != std::string::npos ) 
+      {
+        cDTCsplit  = std::stol( cArg.substr( cArg.find(",")+1 , cArg.length()-1) , nullptr, 10 );
+        std::cout << "DTC split passed from cmd line is " << cDTCsplit << "\n";
+      }
+      if( cArg.find("tkNonant") != std::string::npos ) 
+      {
+        cNonant  = std::stol( cArg.substr( cArg.find(",")+1 , cArg.length()-1) , nullptr, 10 );
+        std::cout << "Tk nonant passed from cmd line is " << cDTCsplit << "\n";
+      }
+    }
+  }
+  int cTotalErrCnt=0;
+
+  // allow for truncation memory check [i.e. missing entries can pass check]
+  bool cTruncation=false;
+  std::string cInputFile_Wires = "emData/LUTs/wires.dat";
+  // to be moved to emulation/wiring script 
+  prepareLUTs(cDTCsplit,  cNonant, cInputFile_Wires ); 
```

Comment:
@sseifeln This line is preventing it from compiling now.

Reply:
Argh! Got distracted and forgot to test again after removing the function .. fixing now.

Reply:
Thanks! Looks good now.